src/shared/contexts/pinneditems.tsx

```diff
@@ -0,0 +1,151 @@
+import React, {
+  createContext,
+  FC,
+  useEffect,
+  useState,
+  memo,
+  useCallback,
+  ReactElement,
+} from 'react'
+import {CLOUD} from '../constants'
+
+let getPinned
+let putPinned
+let deletePinned
+let postPinned
+
+if (CLOUD) {
+  const pinnedItemFunctions = require('src/client/pinnedItemRoutes')
+  getPinned = pinnedItemFunctions.getPinned
+  putPinned = pinnedItemFunctions.putPinned
+  postPinned = pinnedItemFunctions.postPinned
+  deletePinned = pinnedItemFunctions.deletePinned
+}
+
+export interface PinnedItem {
+  orgID: string
+  userID: string
+  id: string
+  createdAt: string
+  updatedAt: string
+  metadata: {
+    [key: string]: string
+  }
+  type: string
+}
+
+export enum PinnedItemTypes {
+  Dashboard = 'dashboard',
+  Cell = 'cell',
+  Notebook = 'notebook',
+  Task = 'task',
+  Alert = 'alert',
+  UploadCSV = 'uploadCsv',
+  UploadLP = 'uploadLP',
+  ClientLibrary = 'clientLibrary',
+}
+export const PinnedItemsContext = createContext(null)
+
+export const getPinnedItems = async () => {
+  const pinnedRes = await getPinned({})
+  if (pinnedRes.status !== 200) {
+    throw new Error('Unable to retrieve pinned items')
+  } else {
+    return pinnedRes.data
+  }
+}
+
+const removePinnedItem = async (id: string) => {
+  await deletePinned({id})
+}
+const addPinnedItem = async (item: Partial<PinnedItem>) => {
+  const pinnedRes = await postPinned({data: item})
+  if (pinnedRes.status !== 200) {
+    throw new Error('Unable to retrieve pinned items')
+  } else {
+    return
+  }
+}
+
+const updatePinnedItem = async (id: string, item: Partial<PinnedItem>) => {
+  await putPinned({id, data: item})
+
+  return
+}
+
+export const pushPinnedItem = async (newItem: Partial<PinnedItem>) => {
+  try {
+    await addPinnedItem(newItem)
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+export const deletePinnedItemByParam = async (param: string) => {
+  try {
+    const items = await getPinnedItems()
+    const toDeleteItem = items.find(item =>
+      Object.values(item.metadata).includes(param)
+    )
+
+    if (toDeleteItem) {
+      await removePinnedItem(toDeleteItem.id)
+    }
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+export const deletePinnedItem = async (itemID: string) => {
+  try {
+    await removePinnedItem(itemID)
+  } catch (err) {
+    console.error(err)
```

Comment:
I'm not sure that logging api errors is secure. I think we would want to handle this differently. See how notebooks does it in src/flows/contex/api.ts. These should probably be shown to the user as notifications, right?